web/models/terms/term.ex

```diff
@@ -21,11 +22,29 @@ defmodule CoursePlanner.Terms.Term do
 
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, [:name, :start_date, :end_date])
-    |> validate_required([:name, :start_date, :end_date])
+    |> cast(params, [:name, :start_date, :end_date, :minimum_teaching_days])
+    |> validate_required([:name, :start_date, :end_date, :minimum_teaching_days])
     |> validate_date_range()
   end
 
+  def validate_minimum_teaching_days(%{valid?: true} = changeset, holidays) do
+    teaching_days = count_teaching_days(changeset)
+    min = Changeset.get_field(changeset, :minimum_teaching_days)
+    if teaching_days - length(holidays) > min do
```

Comment:
what do you think of moving `teaching_days - length(holidays)` to the function `count_teaching_days`?